src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -245,7 +245,7 @@ namespace AirLoopHVACDOAS {
                     auto NodeArray = NodeNames.value();
                     thisMixer.numOfInletNodes = NodeArray.size();
                     int num = 0;
-                    for (auto NodeDOASName : NodeArray) {
+                    for (auto &NodeDOASName : NodeArray) {
```

Comment:
Yes to the reference, otherwise you are making a local copy of the string. No to the `auto`, just use `std::string &`.